VoodooRMI/Transports/I2C/RMII2C.cpp

```diff
@@ -461,3 +460,19 @@ void RMII2C::stopInterrupt() {
         interrupt_source->disable();
     }
 }
+
+OSArray *RMII2C::getConfig() {
+    IOACPIPlatformDevice *acpi_device = OSDynamicCast(IOACPIPlatformDevice, device_nub->getProperty("acpi-device"));
+    if (!acpi_device) {
+        IOLogError("%s::%s Could not retrieve acpi device", getName(), name);
+        return nullptr;
+    };
+
+    OSArray *config = nullptr;
+    OSObject *ret = nullptr;
+    if (acpi_device->evaluateObject("RCFG", &ret) == kIOReturnSuccess)
+        if ((config = OSDynamicCast(OSArray, ret)))
+            config->retain();
+    OSSafeReleaseNULL(ret);
```

Comment:
Probably don't need to retain and then release right afterwards. Your casting, so it's the same object.
I'd just return the OSArray *, even if it's null.
Can check if it's null in RMIBus, and release it there instead.

Reply:
ret is a new OSObject. If we just return the casting result, it couldn't be released later.

Reply:
Yeah I see that now, oops.